api/src/services/xml_generation/value_transformers.py

```diff
@@ -0,0 +1,175 @@
+            f"Currency value must be string or number, got {type(value)}"
+        )
+
+
+def transform_string_case(value: Any, case_type: str) -> str:
+    """Transform string case.
+
+    Args:
+        value: String value to transform
+        case_type: Type of case transformation ("upper", "lower", "title")
+
+    Returns:
+        Transformed string
+
+    Raises:
+        ValueTransformationError: If transformation fails
+    """
+    if not isinstance(value, str):
+        raise ValueTransformationError(
+            f"String transformation requires string input, got {type(value)}"
+        )
+
+    if case_type == "upper":
+        return value.upper()
+    elif case_type == "lower":
+        return value.lower()
+    elif case_type == "title":
+        return value.title()
+    else:
+        raise ValueTransformationError(f"Unsupported case type: {case_type}")
+
+
+def transform_truncate_string(value: Any, max_length: int) -> str:
+    """Truncate string to maximum length.
+
+    Args:
+        value: String value to truncate
+        max_length: Maximum allowed length
+
+    Returns:
+        Truncated string
+
+    Raises:
+        ValueTransformationError: If transformation fails
+    """
+    if not isinstance(value, str):
+        raise ValueTransformationError(
+            f"String truncation requires string input, got {type(value)}"
+        )
+
+    if max_length < 0:
+        raise ValueTransformationError(f"Max length must be non-negative, got {max_length}")
+
+    return value[:max_length] if len(value) > max_length else value
+
+
+# Registry of available transformations
+TRANSFORMATION_REGISTRY: dict[str, Callable[..., Any]] = {
+    "boolean_to_yes_no": transform_boolean_to_yes_no,
+    "currency_format": transform_currency_format,
+    "string_case": transform_string_case,
+    "truncate_string": transform_truncate_string,
+}
+
+
+def apply_value_transformation(value: Any, transformation_config: dict) -> Any:
+    """Apply a value transformation based on configuration.
+
+    Args:
+        value: Value to transform
+        transformation_config: Configuration specifying the transformation
+
+    Returns:
+        Transformed value
+
+    Raises:
+        ValueTransformationError: If transformation fails
+    """
+    transform_type = transformation_config.get("type")
+    if not transform_type:
+        return value  # No transformation specified
+
+    if transform_type not in TRANSFORMATION_REGISTRY:
+        raise ValueTransformationError(f"Unknown transformation type: {transform_type}")
+
+    transformer = TRANSFORMATION_REGISTRY[transform_type]
+
+    # Extract additional parameters
+    params = transformation_config.get("params", {})
+
+    try:
+        return transformer(value, **params)
+    except Exception as e:
+        logger.warning(f"Value transformation failed for {transform_type}: {e}")
+        # Return original value if transformation fails (graceful degradation)
+        return value
```

Comment:
If a transformation fails, we should probably just let that error propagate up. We wouldn't want to partially transform the data as there isn't much value in that. Any transformation error likely is a configurational problem or bug.